src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -227,9 +227,12 @@ internal RegexNode FinalOptimize()
                                     case Alternate:
                                     case Loop:
                                     case Lazyloop:
-                                        var atomic = new RegexNode(Atomic, Options);
-                                        atomic.AddChild(existingChild);
-                                        node.ReplaceChild(node.ChildCount() - 1, atomic);
+                                        if (!existingChild.IsAtomicByParent())
```

Comment:
could you add a comment on why you do this test here? I'm not following

Reply:
Sure, I'll add a comment.  This code is adding an atomic wrapper around a node, e.g. if you have `this|that`, it's changing it to `(?>this|that)`, the benefit of which is better code gen because the backend doesn't need to figure out that the alternation at the end of the expression can't be backtracked into.  But if the regex is actually already `(?>this|that)`, there's no benefit and only downside to making it `(?>(?>this|that))`, so we check to see if there's already an atomic wrapper around the node.